src/hooks/index.js

```diff
@@ -489,147 +578,183 @@ exports.mapPayload = (context) => {
+						context.result.courseGroupId._id;
+				} else {
+					context.result.courseId = context.result.courseId._id;
+				}
 			}
-		}
 
-		if (context.method === 'find' && ((context.result || {}).data || []).length > 0) {
-			context.result.data = context.result.data.filter((lesson) => {
-				if ('courseGroupId' in lesson) {
-					return userIsInThatCourse(user, lesson.courseGroupId, false);
-				}
-				return userIsInThatCourse(user, lesson.courseId, true)
-					|| (context.params.query.shareToken || {}) === (lesson.shareToken || {});
-			});
+			if (
+				context.method === 'find' &&
+				((context.result || {}).data || []).length > 0
+			) {
+				context.result.data = context.result.data.filter((lesson) => {
+					if ('courseGroupId' in lesson) {
+						return userIsInThatCourse(
+							user,
+							lesson.courseGroupId,
+							false,
+						);
+					}
+					return (
+						userIsInThatCourse(user, lesson.courseId, true) ||
+						(context.params.query.shareToken || {}) ===
+							(lesson.shareToken || {})
+					);
+				});
 
-			if (context.result.data.length === 0) {
-				throw new NotFound('There are no lessons that you have access to.');
-			} else {
-				context.result.total = context.result.data.length;
-			}
-			context.result.data.forEach((lesson) => {
-				if ('courseGroupId' in lesson) {
-					lesson.courseGroupId = lesson.courseGroupId._id;
+				if (context.result.data.length === 0) {
+					throw new NotFound(
+						'There are no lessons that you have access to.',
+					);
 				} else {
-					lesson.courseId = lesson.courseId._id;
+					context.result.total = context.result.data.length;
 				}
-			});
+				context.result.data.forEach((lesson) => {
+					if ('courseGroupId' in lesson) {
+						lesson.courseGroupId = lesson.courseGroupId._id;
+					} else {
+						lesson.courseId = lesson.courseId._id;
+					}
+				});
+			}
 		}
-	}
-	return context;
-});
-
-exports.restrictToUsersOwnClasses = (context) => getUser(context).then((user) => {
-	if (testIfRoleNameExist(user, ['superhero', 'administrator', 'teacher'])) {
 		return context;
-	}
-	if (context.method === 'get') {
-		const classService = context.app.service('classes');
-		return classService.get(context.id).then((result) => {
-			const userId = context.params.account.userId.toString();
-			if (!(_.some(result.userIds, (u) => equalIds(u, userId)))
-				&& !(_.some(result.teacherIds, (u) => equalIds(u, userId)))) {
-				throw new Forbidden('You are not in that class.');
+	});
+
+exports.restrictToUsersOwnClasses = (context) =>
+	getUser(context).then((user) => {
+		if (
+			testIfRoleNameExist(user, ['superhero', 'administrator', 'teacher'])
+		) {
+			return context;
+		}
+		if (context.method === 'get') {
+			const classService = context.app.service('classes');
+			return classService.get(context.id).then((result) => {
+				const userId = context.params.account.userId.toString();
+				if (
+					!_.some(result.userIds, (u) => equalIds(u, userId)) &&
```

Comment:
Codacy found an issue: ['&&' should be placed at the beginning of the line.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5994445)